### PR TITLE
ZFIN-8218: partial refactor of HibernatePublicationRepository

### PR DESCRIPTION
**Pull request by rtaylorzfin**
This is a partial refactor of HibernatePublicationRepository.

Converted some appended strings into text blocks
Converted Query objects to parameterized Query objects
Converted some queries to the JPA Criteria Builder style
Removed methods that seem to be unused
Re-ordered methods in HibernatePublicationRepository to match order in PublicationRepository
Changed public methods to private methods if they are only used in HibernatePublicationRepository (and moved them to the bottom of the file)
Added test file: PublicationRepositoryRefactorTest to test methods that were converted (if not covered by PublicationRepositoryTest)
Removed tests for removed methods

### CODE REVIEW

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java [hunk 1]
```diff
@@ -83,148 +83,48 @@ public class HibernatePublicationRepository extends PaginationUtil implements Pu
...
         Session session = HibernateUtil.currentSession();
-        String hql = "SELECT distinct publication FROM Publication publication, ExpressionExperiment exp, ExpressionResult res   " +
-            "WHERE (res.entity.superterm.zdbID = :aoZdbID OR" +
-            "       res.entity.subterm.zdbID = :aoZdbID) " +
-            "AND publication = exp.publication " +
-            "AND res.expressionExperiment = exp " +
-            "AND exp.gene.zdbID = :zdbID " +
-            "AND res.expressionFound = :expressionFound ";
+        String hql = """
+            SELECT distinct publication FROM Publication publication, ExpressionExperiment exp, ExpressionResult res  \s
```
Comment (cmpich) [inline source/org/zfin/publication/repository/HibernatePublicationRepository.java:89]: What is the \s character at the end of this line?

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java [hunk 2]
```diff
@@ -821,19 +352,19 @@ public int getNumPublicationsWithFiguresPerGenotypeAndAnatomy(Genotype genotype,
...
-            "   res.expressionExperiment = exp AND " +
-            "   (res.entity.superterm = :aoTerm OR res.entity.subterm = :aoTerm) AND " +
-            "   xpatfig.expressionResult = res AND " +
-            "   xpatfig.figure = fig AND " +
-            "   res.expressionFound = :expressionFound AND " +
-            "   exp.fishExperiment = fishox AND " +
-            "   fishox.standardOrGenericControl = :condition AND " +
-            "   fishox.fish.genotype = geno AND " +
-            "   fishox.fish.genotype.wildtype = :isWildtype ";
+                "     FishExperiment fishox, ExpressionResultFigure xpatfig, Genotype geno " +
```
Comment (cmpich) [inline source/org/zfin/publication/repository/HibernatePublicationRepository.java:355]: text block would be nice here...

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java [hunk 3]
```diff
@@ -843,354 +374,198 @@ public List<Figure> getFiguresByGeneAndAnatomy(Marker marker, GenericTerm anatom
...
-        try {
-            Session session = HibernateUtil.currentSession();
-            Criteria criteria = session.createCriteria(Journal.class);
-            criteria.add(Restrictions.eq("name", journalTitle));
-            return (Journal) criteria.uniqueResult();
-        } catch (Exception e) {
-            logger.error("failed to get journal title[" + journalTitle + "] returning null", e);
-            return null;
+        List<String> publicationIDs = HibernateUtil.currentSession()
+                .createSQLQuery(getCommonPublicationSQL(zdbID))
```
Comment (cmpich) [inline source/org/zfin/publication/repository/HibernatePublicationRepository.java:381]: isn't this still the deprecated method?

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java [hunk 4]
```diff
@@ -843,354 +374,198 @@ public List<Figure> getFiguresByGeneAndAnatomy(Marker marker, GenericTerm anatom
...
+                "FishExperiment fishox, Genotype geno, ExpressionResultFigure xpatfig " +
+                "where geno.zdbID = :genoID AND " +
+                "      fishox.fish.genotype = geno AND " +
+                "   res.expressionExperiment = exp AND " +
+                "   xpatfig.expressionResult = res AND " +
+                "   xpatfig.figure = figure AND " +
+                "   exp.antibody is null AND " +
+                "   exp.fishExperiment = fishox  " +
+                "order by figure.orderingLabel    ";
+        Query query = session.createQuery(hql);
```
Comment (cmpich) [inline source/org/zfin/publication/repository/HibernatePublicationRepository.java:461]: is this using org.hibernate.query.Query? 
org.hibernate.Query is deprecated.

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java [hunk 5]
```diff
@@ -2139,6 +1372,15 @@ public long getPhenotypeCount(Publication publication) {
         return getCount(sql, publication.getZdbID());
     }
 
+    public long getPhenotypeAlleleCount(Publication publication) {
+        String sql = "\tselect count(distinct geno_zdb_id)" +
+                "\tfrom   record_attribution, genotype" +
```
[the remark this answers is not in the document]
Comment (rtaylorzfin) [reply source/org/zfin/publication/repository/HibernatePublicationRepository.java:1377]: removed. Future refactor should change to text blocks

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java (hunk 4 above)
Comment (rtaylorzfin) [reply source/org/zfin/publication/repository/HibernatePublicationRepository.java:461]: updated to use the new Query interface.

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java (hunk 3 above)
Comment (rtaylorzfin) [reply source/org/zfin/publication/repository/HibernatePublicationRepository.java:381]: changed to createNativeQuery

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java (hunk 2 above)
Comment (rtaylorzfin) [reply source/org/zfin/publication/repository/HibernatePublicationRepository.java:355]: added text block

File: source/org/zfin/publication/repository/HibernatePublicationRepository.java (hunk 1 above)
Comment (rtaylorzfin) [reply source/org/zfin/publication/repository/HibernatePublicationRepository.java:89]: I think it was added automatically by the intellij conversion to text block. It must be to preserve white space at the end of a line. It was unnecessary, so I removed it.

**Review by cmpich — APPROVED** [review]
Looks good.

**Pull request merged**